src/librustc/query/mod.rs

```diff
@@ -653,6 +684,9 @@ rustc_queries! {
     }
 
     TypeChecking {
+        query all_local_trait_impls(key: CrateNum) -> &'tcx BTreeMap<DefId, Vec<hir::HirId>> {
```

Comment:
Long-term, we should probably turn this (and other similar things if there are any) into a regular query by iterating through all HIR items or something.